src/dune_lang/blang.ml

```diff
@@ -47,68 +47,74 @@ module Op = struct
   ;;
 end
 
-type t =
+type 'string ast =
```

Comment:
Why do you need this type? Can't you just inline it into `Ast.t`?

Reply:
In the mli you can move the Ast module type up with the same order as the ml file which would let you write `type t = String_with_vars.t Ast.t`

Reply:
It's done this way so that the constructors for `Blang.t` are defined inside the `Blang` module (so you can write `Blang.And` for example). I'll add a comment explaining this.

Reply:
Added a comment. We could also introduce functions to use instead of constructors and move the constructors to `Blang.Ast`. Let me know what you think.

Reply:
I see why you did it now.